swo_aws_extension/flows/steps/setup_context.py

```diff
@@ -0,0 +1,63 @@
+import logging
+
+from mpt_extension_sdk.mpt_http.mpt import update_order
+
+from swo_aws_extension.aws.client import AWSClient
+from swo_aws_extension.aws.config import Config
+from swo_aws_extension.constants import PhasesEnum
+from swo_aws_extension.flows.order import InitialAWSContext
+from swo_aws_extension.flows.steps.base import BasePhaseStep
+from swo_aws_extension.flows.steps.errors import ConfigurationStepError, SkipStepError
+from swo_aws_extension.parameters import get_phase, set_phase
+
+logger = logging.getLogger(__name__)
+
+
+class SetupContext(BasePhaseStep):
+    """Initial setup context step."""
+
+    def __init__(self, config: Config, role_name: str) -> None:
+        self._config = config
+        self._role_name = role_name
+
+    def pre_step(self, context: InitialAWSContext) -> None:
+        """Validate if step should run."""
+        if not context.pm_account_id:
+            raise ConfigurationStepError(
+                "SetupContextError - PMA account is required to setup AWS Client in context"
+            )
+
+    def process(self, context: InitialAWSContext) -> None:
+        """Setup AWS client in context."""
+        self._setup_aws(context)
```

Comment:
Regarding my comment about AWS credentials validation... you can see we are instantiating the AWSClient twice, why is that? Does it make sense to have an aws_client without validating whether the credentials are correct? I
If we don't want to follow the AWSService approach to isolate this logic, I truly believe that the setup context should be the owner of adding and validating everything we need across the pipeline